### ISSUE TITLE
Upgrade to hitch from 1.4.4 to 1.4.5 in Centos7 breaks multi-frontend configuration

### ISSUE THREAD

**Issue by bago** [varnish/hitch#191]
Hi,

I had a working deploy using hitch 1.4.4 with 3 frontends bound on 3 different IPs.

Each frontend declares its own certificates (a main wildcard certificate for non-SNI and additional certificates for SNI enabled clients).

After I updated to hitch 1.4.5 it seems all the frontends started sending all the wildcard certificates for the 3 frontends in a random order breaking most validations.

I looked at the changelog but I don't see any needed configuration change or anything that pointed me to some explanation for this weird behaviour.

I downgraded to 1.4.4 via rpm.

I'm sorry I don't have more informations / debug info for this issue, but I thought it was better to share this ASAP so if other users have similar issues upgrading will be able to append here.

**Comment by HLeithner** [comment]
Same problem as #158

**Comment by bago** [comment]
The description seems the same. Unfortunately the "ocsp-dir" configuration is already commented in my conf, so I can't try the "solution" described in the #158 .

Here is my conf:
```
backend = "[127.0.0.1]:6081"
pem-file = ""
tls-protos = TLSv1.0 TLSv1.1 TLSv1.2
ciphers = "-MEDIUM:-LOW:-aNULL:-eNULL:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA:ECDHE-ECDSA-AES128-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE
prefer-server-ciphers = on
ssl-engine = ""
workers = 1
backlog = 100
keepalive = 3600
chroot = ""
user = "hitch"
group = "hitch"
quiet = on
syslog = on
syslog-facility = "daemon"
daemon = off
write-ip = off
write-proxy-v1 = off
write-proxy-v2 = on
proxy-proxy = off
sni-nomatch-abort = off
frontend = {
  host = "IP1"
  port = "443"
  pem-file = "...."
  pem-file = "...."
  ....
}
frontend = {
  host = "IP2"
  port = "443"
  pem-file = "...."
  pem-file = "...."
  ....
}
frontend = {
  host = "IP3"
  port = "443"
  pem-file = "...."
  pem-file = "...."
  ....
}
```

**Comment by bago** [comment]
I just read the changelog and see that since 1.4.3 the ocsp is enabled by default and that I'm responsible for creating the folder. So I'll check this ASAP.

**Comment by HLeithner** [comment]
It seams that the ocsp wasn't the real problem, I tried to find it but I wasn't able to find the problem...

**Comment by bago** [comment]
it sounds like a major issues... it is weird that we are the only 2 users reporting it if the issue was not only in the latest (recent) centos 7 release but was there since january.

**Comment by klenan** [comment]
When defining frontends as blocks, hitch would always bind to `::` completely ignoring the specified `host`, hence all certificates would be offered to everyone.

https://github.com/varnish/hitch/pull/206/commits/55604ef48bfaee1cc5187c9d53caad74bb27e4e6 fixes the issue #191 and #158, the diff is self explanatory.
